students/roviram/Lesson04/Mailroom_Part_Two.py

```diff
@@ -0,0 +1,135 @@
+        {"Name": "Pepper", "Donation Amount": [5, 20, 100]},
+        {"Name": "Vicky Gonzalez", "Donation Amount": [600, 9000, 10]},
+        {"Name": "Cristina Rovira", "Donation Amount": [25, 50, 75]}
+]
+"""
+for dictionary in donors:
+    for dict_values in dictionary.values():
+        print(dict_values)
+        if dict_values == "Miguel Rovira":
+            print("yes") 
+"""
+
+# Processing
+def gen_stats(donors):
+    """Generating stats of the donors"""
+    gen_list = []
+    final_list = donors
+    for donor in donors:
+        donations = donor["Donation Amount"]
+        total = sum(donations)
+        number_of_donations = len(donations)
+        avg = round(total / number_of_donations, 2)
+        stats = [donor["Name"], total, number_of_donations, avg]
+        gen_list.append(stats)
+    return gen_list
+
+def create_report(report_donors):
+    """Creating a report of the mail room donors"""
+    stats_list = gen_stats(report_donors)
+    stats_list.sort(key=lambda stats_list: stats_list[1], reverse=True)
+    print("{:<20} | {:^15} | {:>15} | {:>15}".format("Name of Donor", "Total Donation Amount", "Number of Donations", "Average Gift Amount"))
+    print("-"*89)
+    for index, donor in enumerate(stats_list):
+        print(str(index + 1) + ".", "{:<20}${:^15}{:>15}\t\t\t\t{}{:>15}".format(donor[0], donor[1], donor[2], "$", donor[3]))
+
+def intro():
+    """Welcoming the user to the application"""
+    # Intro to the application
+    print("Hello! Welcome to the mailroom! We are excited you are apart of our team.\n")
+
+def thank_you():
+    """Thanking the donor for their donor and adding a new donor if they are new"""
+
+    while True:
+        list_answer = input("\nThank you notes are always appreciated!\n"
+                            "Want to see the list of donors to send a 'Thank You' too? (type 'Yes' or 'No ') ")
+        # Cleaning up the Users Answer
+        if list_answer.upper().strip() == 'YES':
+            for donor in donors:
+                print(donor)
+
+        # Prompt Donor Name
+        donor_name = input("\nPlease enter the donor name: ")
+
+        print("You typed:", donor_name)
+        match = False
+        for dict_donors in donors:
+            if dict_donors["Name"].lower() == donor_name.lower():
+                print("Cool, I found that donor")
+                match = True
+        #Checking if match is true
+        if match:
+            pass
+        else:
+            print("I didn't find that donor, I am adding them to our database now.")
+            new_donor = {"Name": donor_name, "Donation Amount": []}
+            donors.append(new_donor)
+
+        donation_amount = float(input("\nHow much did they donate?: "))
+        for dict_donors in donors:
+            if dict_donors["Name"].lower() == donor_name.lower():
+                dict_donors["Donation Amount"].append(donation_amount)
+
+        print()
```

Comment:
Instead of having an extra print, you can use a "\n" in the following print to add a new line